packages/client/components/EditorLinkChanger/EditorLinkChangerDraftjs.tsx

```diff
@@ -37,11 +51,49 @@ const EditorLinkChangerDraftjs = (props: Props) => {
   } = props
   useTaskChild('editor-link-changer')
   const handleSubmit = ({text, href}: {text: string; href: string}) => {
+    const keepSelection = true
+    // don't include leading and trailing whitespace in the link text
+    const nonWhitespaceFromStart = text.search(/\S/)
+    const nonWhitespaceFromEnd = text.search(/\s*$/)
+    const startStr = nonWhitespaceFromStart === -1 ? '' : text.slice(0, nonWhitespaceFromStart)
+    const endStr = nonWhitespaceFromEnd === -1 ? '' : text.slice(nonWhitespaceFromEnd)
+    const trimmedText = text.trim() ? text.trim() : text
     const focusedEditorState = EditorState.forceSelection(editorState, selectionState)
-    const nextEditorState = completeEntity(focusedEditorState, 'LINK', {href}, text, {
-      keepSelection: true
+    let newEditorState = focusedEditorState
+    // For whitespace only links
+    if (text.trim()) {
+      const contentState = focusedEditorState.getCurrentContent()
+      const expandedSelectionState = keepSelection
```

Comment:
-1 this is always `true`

Reply:
Here I mean, because `keepSelection` is always true, we can just write
```ts
      const expandedSelectionState = focusedEditorState.getSelection()
```
directly

Reply:
yeah